File: js/js.translator/testData/box/nestedTypes/outerNative.js
function A(x) {
    this.x = x;
}
A.prototype = {
    foo : function() {
        return this.x;
    }
};

A.B = function(konstue) {
    this.konstue = konstue;
};
A.B.prototype = {
    bar : function() {
        return 10000 + this.konstue;
    }
};

A.C = function(outer, konstue) {
    this.outer = outer;
    this.konstue = konstue;
};
A.C.prototype = {
    bar : function() {
        return this.outer.foo() + this.konstue + 10000;
    },
    dec : function() {
        this.outer.x--;
    }
};

File: js/js.translator/testData/box/nestedTypes/outerNative.kt
// EXPECTED_REACHABLE_NODES: 1281
package foo

external class A(x: Int) {
    var x: Int
        get() = definedExternally
        set(konstue) = definedExternally

    fun foo(): Int = definedExternally

    class B(konstue: Int) {
        konst konstue: Int

        fun bar(): Int = definedExternally
    }
}

fun box(): String {
    var b = A.B(23)
    if (b.bar() != 10023) return "failed1: ${b.bar()}"

    return "OK"
}

